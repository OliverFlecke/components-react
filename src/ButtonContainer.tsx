import React, { FC } from 'react';

type Position = 'Left' | 'Right' | 'Center';

export interface ButtonContainerProps {
	position?: Position;
	className?: string;
	children: any;
}

const ButtonContainer: FC<ButtonContainerProps> = ({
	position,
	className,
	children,
}) => (
	<div
		className={`w-full flex flex-row rounded-md space-x-4 p-4 bg-gray-50 dark:bg-gray-900 ${getPositionClass(
			position
		)} ${className}`}
	>
		{children}
	</div>
);

export default ButtonContainer;

function getPositionClass(position?: Position) {
	switch (position) {
		case 'Left':
			return 'justify-start';
		case 'Center':
			return 'justify-center';

		case 'Right':
		default:
			return 'justify-end';
	}
}
