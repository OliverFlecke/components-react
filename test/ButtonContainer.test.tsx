import React from 'react';
import renderer from 'react-test-renderer';
import ButtonContainer, { Position } from '../src/ButtonContainer';

describe('ButtonContainer', () => {
	test.each(['Left', 'Right', 'Center'] as Position[])(
		'Render snapshot of container in each possible position',
		(position: Position) => {
			const tree = renderer
				.create(
					<ButtonContainer position={position}>Some content</ButtonContainer>
				)
				.toJSON();
			expect(tree).toMatchSnapshot();
		}
	);

	test('With custom class', () => {
		const tree = renderer.create(
			<ButtonContainer className="custom-class">Any content</ButtonContainer>
		);
		expect(tree.toJSON()).toMatchSnapshot();
	});
});
