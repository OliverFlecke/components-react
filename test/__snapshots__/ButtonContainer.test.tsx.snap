// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContainer Render snapshot of container in each possible position 1`] = `
<div
  className="w-full flex flex-row rounded-md space-x-4 p-4 bg-gray-50 dark:bg-gray-900 justify-start"
>
  Some content
</div>
`;

exports[`ButtonContainer Render snapshot of container in each possible position 2`] = `
<div
  className="w-full flex flex-row rounded-md space-x-4 p-4 bg-gray-50 dark:bg-gray-900 justify-end"
>
  Some content
</div>
`;

exports[`ButtonContainer Render snapshot of container in each possible position 3`] = `
<div
  className="w-full flex flex-row rounded-md space-x-4 p-4 bg-gray-50 dark:bg-gray-900 justify-center"
>
  Some content
</div>
`;

exports[`ButtonContainer With custom class 1`] = `
<div
  className="w-full flex flex-row rounded-md space-x-4 p-4 bg-gray-50 dark:bg-gray-900 justify-end custom-class"
>
  Any content
</div>
`;
